src/ape_ethereum/ecosystem.py

```diff
@@ -95,8 +96,8 @@ def decode_calldata(self, abi: ABI, raw_data: bytes) -> Any:
         try:
             return abi_decode(output_types, raw_data)
 
-        except InsufficientDataBytes as e:
-            raise Exception("Output corrupted") from e
+        except InsufficientDataBytes as err:
+            raise ContractCallError("Output corrupted") from err
```

Comment:
Does `decode_calldata()` only have to do with contract calls?
I am trying to make sense of the all the use-cases of this method so I know if this is the best custom exception type to use.

Reply:
Contract transactions can also return data that needs to be decoded (not sure if this is implemented yet)

I would say that `DecodingError` might be a better name

Reply:
Ok, so I:

1. made a base exception `ContractError`.
2. made `TransactionError` and `ContractCallError` subclasses of `ContractError`
3. made new error `DecodingError` that is also a subclass of `ContractError`

I think that models the relationships a little better here.